editortask.cpp

```diff
@@ -449,16 +449,19 @@ void EditorTask::menuRun()
 		{
 			fwrite(buffer.c_str(), buffer.length(), 1, f);
 			fclose(f);
-			
+		}
+		
+		if (!ferror(f))
```

Comment:
I'm not sure whether this is implemented fully in ndless - can you check the return value of fwrite instead?

Reply:
Hmm it worked fine when compiled with the latest ndless sdk on both my calculator and firemu but if you really want to, I can

Reply:
```suggestion
				FILE *f = fopen("/documents/tmp.py", "wb");
		if(f)
		{
			if(fwrite(buffer.c_str(), buffer.length(), 1, f))
				runFile("/documents/tmp.py");
			else
			{
				dialog_task.showMessage("Could not save 'tmp.py'!");
				return;
			}
			fclose(f);
			remove("/documents/tmp.py");
		}
		else
			dialog_task.showMessage("Could not save 'tmp.py'!");
```

I don't know how to do it in a cleaner way tho...

Reply:
Now it doesn't close the file if `fwrite` fails, what about:

```
FILE *f = fopen("/documents/tmp.py", "wb");
if(f && fwrite(buffer.c_str(), buffer.length(), 1, f))
    runFile("/documents/tmp.py");
else
    dialog_task.showMessage("Could not save 'tmp.py'!");

if(f)
{
    fclose(f);
    remove("/documents/tmp.py");
}
```

Reply:
Way cleaner, thank you very much